app/logical/source/extractor/mihuashi.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+# @see Source::URL::Mihuashi
+module Source
+  class Extractor
+    class Mihuashi < Source::Extractor
+      def image_urls
+        if parsed_url.full_image_url.present?
+          [parsed_url.full_image_url]
+        elsif parsed_url.image_url?
+          [parsed_url.to_s]
+        elsif work.present?
+          image_url = work[:url]
+          [Source::URL.parse(image_url).try(:full_image_url) || image_url]
+        elsif stall.present?
+          [
+            stall[:cover_url],
+            *stall[:example_images].to_a,
+          ].uniq.map do |url|
+            Source::URL.parse(url).try(:full_image_url) || url
+          end
+        elsif project.present?
+          project[:example_images].to_a.pluck(:url).map do |url|
+            Source::URL.parse(url).try(:full_image_url) || url
+          end
+        elsif character_card.present?
+          [
+            character_card[:image_url],
+            *character_card[:example_images].to_a.pluck(:url),
+          ].uniq.map do |url|
+            Source::URL.parse(url).try(:full_image_url) || url
+          end
+        else
+          []
+        end
+      end
+
+      def profile_url
+        "https://www.mihuashi.com/users/#{Danbooru::URL.escape(username)}?role=painter" if username.present?
+      end
+
+      def account_url
+        "https://www.mihuashi.com/profiles/#{user_id}?role=painter" if user_id.present?
+      end
+
+      def profile_urls
+        [profile_url, account_url].compact
+      end
+
+      def display_name
+        username&.strip
```

Comment:
`display_name` and `username` should not be the same thing. `display_name` is for sites like Twitter or Pixiv where users have two different names, a username that uniquely identifies the account and a display name that doesn't. If they only have a username, then `display_name` should be nil.